### PR TITLE
Fix Rich-Text & Image Upload in FRESHLY Posted Comments

### LINKED ISSUE #9072: [Bug] Attach Rich-Text & Image Upload Events on FRESH, Newly Created Comment Forms
Clicking on rich-text (bold, italic, etc.) buttons and image upload (both select & drag-and-drop) will not work on FRESH newly-created comment forms created by posting a comment. Image of comment form for more context:
Screen Shot 2021-01-24 at 7 18 32 PM

[Video of This Bug in Action!](https://user-images.githubusercontent.com/4361605/105657600-9d6aad80-5e79-11eb-8fc0-9e31f9c727e0.mov)

This happens because the eventHandlers in `editorToolbar.js` that attach `click` and `fileupload` events are only attached at the moment that the `document` is loaded. So new elements that are created after `document.load` do not have these events attached:
https://github.com/publiclab/plots2/blob/38e37eb20602d799ab935e4e281f32782e856fda/app/assets/javascripts/editorToolbar.js#L41-L48

Working on a fix for this one!

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #9069 for more context)

### PR DESCRIPTION
**Pull request by noi5e**
Fixes #9072

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #9069 for more context)

### CODE REVIEW

File: app/assets/javascripts/editor.js [hunk 1]
```diff
@@ -132,9 +132,6 @@ class Editor {
       this.textAreaElement.val(this.textAreaValue+'\n\n'+this.templates[template])
     }
   }
-  generate_preview(id,text) {
-    $('#' + id)[0].innerHTML = marked(text)
-  }
```
Comment (noi5e) [inline app/assets/javascripts/editor.js:137]: Unrelated changes in this file (sorry)! Deleted this unused code.

File: app/views/comments/create.js.erb [hunk 2]
```diff
@@ -3,8 +3,21 @@
...
+const replyButton = $('#dropzone-small-reply-<%= @comment.cid %>');
+const editButton = $('#dropzone-small-edit-<%= @comment.cid %>');
+const replyFormOptions = getFileUploadOptions(replyFormBody, false);
+const editFormOptions = getFileUploadOptions(editFormBody, false);
+const replyButtonOptions = getFileUploadOptions(replyButton, true);
+const editButtonOptions = getFileUploadOptions(editButton, true);
+replyFormBody.fileupload(replyFormOptions);
+editFormBody.fileupload(editFormOptions);
+replyButton.fileupload(replyButtonOptions);
+editButton.fileupload(editButtonOptions);
```
Comment (noi5e) [inline app/views/comments/create.js.erb:19]: I think `create.js.erb` runs when `comments/create` is called from the controller.

I had a choice between putting this code here, or in `/app/assets/javascripts/comment.js`. The second choice (where `ajax:success` is triggered on adding a comment) seemed more intuitive, but also more complicated to rewrite. Putting this here for now.

File: app/views/editor/_editor.html.erb [hunk 3]
```diff
@@ -28,8 +28,23 @@
...
+        </span>
+        <!-- http://stackoverflow.com/questions/11235206/twitter-bootstrap-form-file-element-upload-button -->
+        <label for="fileinput-choose-one-main">
+          <input id="fileinput-choose-one-main" type="file" name="image[photo]" style="display:none;" />
+          <a style="cursor: pointer;" class="d-none d-md-inline text-underline"><%= translation('comments._form.choose_one') %></a>
+          <span class="d-md-none">
+            <i class="fa fa-upload"></i>
+            <a><%= translation('comments._form.upload_image') %></a>
+          </span>
+        </label>
```
Comment (noi5e) [inline app/views/editor/_editor.html.erb:47]: This is an unrelated change (sorry again), for some reason the imagebar text ("Drag and drop an image, or choose one to upload") wasn't appearing in the legacy editor pages. It was just a blank grey spot.

**Comment by noi5e** [comment]
@jywarren Do you have the ability to override codeclimate? I can't do it. Would prefer not to take the suggestions in this case.

Otherwise this is ready to merge!

**Comment by jywarren** [comment]
Yes i can do that tomorrow morning! Thanks!

File: app/assets/javascripts/editorToolbar.js [hunk 4]
```diff
@@ -7,126 +7,130 @@
...
-          console.log(e);
-        },
-        progressall: function (e, data) {
-          const closestProgressBar = $("#image-upload-progress-bar-" + $E.commentFormID);
-          return progressAll(closestProgressBar, data);
-        }
-    });
+    const isSmallDropzone = $(this).hasClass("dropzone-small");
+    const fileUploadOptions = getFileUploadOptions($(this), isSmallDropzone);
+    $(this).fileupload(fileUploadOptions);
```
Comment (Sagarpreet) [inline app/assets/javascripts/editorToolbar.js:131]: 👍

**Review by Sagarpreet — APPROVED** [review]

File: app/views/comments/create.js.erb (hunk 2 above)
Comment (jywarren) [reply app/views/comments/create.js.erb:19]: That note on when/where it is run may be good to include in the file itself in a header comment, how does that sound? Just thinking to help future devs. Thanks!!!

**Review by jywarren — APPROVED** [review]
Thanks @noi5e and reviewers!!!!

**Pull request merged**